ring/lifecycler.go

```diff
@@ -81,7 +82,7 @@ func (cfg *LifecyclerConfig) RegisterFlagsWithPrefix(prefix string, f *flag.Flag
 		panic(fmt.Errorf("failed to get hostname %s", err))
 	}
 
-	cfg.InfNames = []string{"eth0", "en0"}
+	cfg.InfNames = netutil.PrivateNetworkInterfaces(log.NewLogfmtLogger(os.Stdout))
```

Comment:
Take the logger in input to `RegisterFlagsWithPrefix()` (as 3rd parameter) and then in `RegisterFlags()` as 2nd parameter. It's a pattern we already use elsewhere and there's an interface defined for that (see `RegistererWithLogger` interface in `flagext/register.go`).